include/seqan3/argument_parser/validators.hpp

```diff
@@ -485,6 +485,98 @@ struct default_validator
+
+    /*!\brief Constructing from two validators.
+     * \param[in] vali1_ Some validator to be chained to vali2_.
+     * \param[in] vali2_ Another validator to be chained to vali1_.
+     */
+    validator_chain_adaptor(validator1_type const & vali1_, validator2_type const & vali2_) :
+        vali1{vali1_}, vali2{vali2_}
+    {}
+
+    //!\brief The destructor.
+    ~validator_chain_adaptor() = default;
+    //!\}
+
+    /*!\brief Calls the operator() of each validator on the value cmp.
+     * \param[in] cmp The value to validate.
+     *
+     * This function delegates to the validation of both of the chained validators
+     * by calling their operator() one after the other. The behaviour depends on
+     * the chained validators which may throw on input error.
+     */
+    void operator()(value_type const & cmp) const
+    {
+        vali1(cmp);
+        vali2(cmp);
+    }
+
+    //!\brief Returns a message that can be appended to the (positional) options help page info.
+    std::string get_help_page_message() const
+    {
+        return detail::to_string(vali1.get_help_page_message(), " ", vali2.get_help_page_message());
+    }
+
+private:
+    //!\brief The first validator in the chain.
+    validator1_type vali1;
+    //!\brief The second validator in the chain.
+    validator2_type vali2;
+};
+
 } // namespace detail
 
+/*!\brief Enables the chaining of validators.
+ *!\ingroup validators
+ * \tparam validator1_type The type of the fist validator;
+ *                            Must satisfy the seqan3::validator_concept and the
+ *                            same value_type as the second validator type.
+ * \tparam validator2_type The type of the second validator;
+ *                            Must satisfy the seqan3::validator_concept  and the
+ *                            same value_type as the fist validator type.
+ * \param[in] vali1 The first validator to chain.
+ * \param[in] vali2 The second validator to chain.
+ * \returns A new validator that tests a value for both vali1 and vali2.
+ *
+ * \details
+ *
+ * The operator+() is the AND operation for two validators, which means that a
+ * value must pass both validators in order to be accepted by the new validator.
+ *
+ * For example you may want a file name that only accepts absolute paths but
+ * also must have one out of a some given file extensions.
+ * For this purpose you can chain a seqan3::regex_validator to a
+ * seqan3::file_ext_validator like this:
+ *
+ * \snippet test/snippet/argument_parser/validators_chaining.cpp all
+ *
+ * You can chain as many validators as you want which will be evaluated one after
+ * the other from left to right (first to last).
+ */
+template <validator_concept validator1_type, validator_concept validator2_type>
+    requires std::Same<typename validator1_type::value_type, typename validator2_type::value_type>
+auto operator+(validator1_type const & vali1, validator2_type const & vali2)
```

Comment:
Here you should use forwarding references and forward the objects into the operator.